File: src/main/java/io/sch/historyscan/domain/contexts/analysis/history/CodeBaseHistoryCommitFile.java
package io.sch.historyscan.domain.contexts.analysis.history;

public record CodeBaseHistoryCommitFile(
        FileInfo fileInfo,
        int currentNbLines,
        int nbAddedLines,
        int nbDeletedLines,
        int nbModifiedLines) {

    public int cloc() {
        return nbAddedLines + nbDeletedLines + nbModifiedLines;
    }

    public String path() {
        return fileInfo.path();
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/analysis/history/CodeBaseHistoryCommitFile.kt
package io.sch.historyscan.domain.contexts.analysis.history

class CodeBaseHistoryCommitFile(
    val fileInfo: FileInfo,
    val currentNbLines: Int,
    val nbAddedLines: Int,
    val nbDeletedLines: Int,
    val nbModifiedLines: Int
) {
    fun cloc(): Int {
        return nbAddedLines + nbDeletedLines + nbModifiedLines
    }

    fun path(): String {
        return fileInfo.path
    }
}
